Sources/MapboxCoreNavigation/NavigationService.swift

```diff
@@ -308,8 +308,8 @@ public class MapboxNavigationService: NSObject, NavigationService {
         
         let routerType = routerType ?? DefaultRouter.self
         _router = routerType.init(alongRouteAtIndex: routeIndex, in: routeResponse, options: routeOptions, directions: self.directions, dataSource: self)
-        NavigationSettings.shared.distanceUnit = routeOptions.locale.usesMetric ? .kilometer : .mile
-        
+        NavigationSettings.shared.distanceUnit = .init(routeOptions.distanceMeasurementSystem)
```

Comment:
Is this still necessary, now that `NavigationRouteOptions.distanceMeasurementSystem` is set to match `NavigationSettings.distanceUnit` by default? If the developer goes out of their way to change `distanceMeasurementSystem` away from the default, should we keep the UI in sync, or should we allow the developer to configure voice instructions in a different unit than the UI?

Reply:
https://github.com/mapbox/mapbox-navigation-ios/issues/3588